src/nlr/Layer.cpp

```diff
@@ -879,38 +879,43 @@ void Layer::computeSymbolicBoundsForWeightedSum()
         const Layer *sourceLayer = _layerOwner->getLayer( sourceLayerEntry.first );
 
         /*
-          Perform the multiplication.
+          Perform the multiplication, don't change pre calucalted bounds for
+          eliminated variables
 
           newUB = oldUB * posWeights + oldLB * negWeights
           newLB = oldUB * negWeights + oldLB * posWeights
         */
 
-        for ( unsigned i = 0; i < _inputLayerSize; ++i )
-        {
-            for ( unsigned j = 0; j < _size; ++j )
-            {
-                if ( _eliminatedNeurons.exists( j ) )
-                    continue;
-
-                for ( unsigned k = 0; k < sourceLayerSize; ++k )
-                {
-                    _symbolicLb[i * _size + j] +=
-                        sourceLayer->getSymbolicUb()[i * sourceLayerSize + k] *
-                        _layerToNegativeWeights[sourceLayerIndex][k * _size + j];
-
-                    _symbolicLb[i * _size + j] +=
-                        sourceLayer->getSymbolicLb()[i * sourceLayerSize + k] *
-                        _layerToPositiveWeights[sourceLayerIndex][k * _size + j];
-
-                    _symbolicUb[i * _size + j] +=
-                        sourceLayer->getSymbolicUb()[i * sourceLayerSize + k] *
-                        _layerToPositiveWeights[sourceLayerIndex][k * _size + j];
-
-                    _symbolicUb[i * _size + j] +=
-                        sourceLayer->getSymbolicLb()[i * sourceLayerSize + k] *
-                        _layerToNegativeWeights[sourceLayerIndex][k * _size + j];
+        Map<unsigned, double> _eliminatedNeuronsLB;
+        Map<unsigned, double> _eliminatedNeuronsUB;
```

Comment:
the underscore notation _name is only used for members of the class, not for local variables